### PR TITLE
initial support for query parameters

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Note that the path* operation groups have been omitted from the package as the generated code doesn't compile.
Not all of the tests have been implemented, and a few being added are skipped due to failures that need investigation.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Issue tracking query params https://github.com/Azure/autorest.go/issues/265

File: src/generator/protocol/operations.ts [hunk 1]
```diff
@@ -83,6 +83,70 @@ class methodSig implements MethodSig {
...
+      return `"${constSchema.value.value}"`;
+    case SchemaType.Date:
+    case SchemaType.DateTime:
+      return `${param.language.go!.name}.String()`;
+    case SchemaType.Integer:
+      imports.add('strconv');
+      const intSchema = <NumberSchema>param.schema;
+      let intParam = param.language.go!.name;
+      if (intSchema.precision === 32) {
+        intParam = `int64(${intParam})`;
```
Comment (catalinaperalta) [inline src/generator/protocol/operations.ts]: Just checking, why is this int64 and not int32?

File: src/generator/protocol/operations.ts [hunk 2]
```diff
@@ -83,6 +83,70 @@ class methodSig implements MethodSig {
...
+      }
+      return `strconv.FormatInt(${intParam}, 10)`;
+    case SchemaType.Number:
+      imports.add('strconv');
+      const numberSchema = <NumberSchema>param.schema;
+      let floatParam = param.language.go!.name;
+      if (numberSchema.precision === 32) {
+        floatParam = `float64(${floatParam})`;
```
Comment (catalinaperalta) [inline src/generator/protocol/operations.ts]: Same question as the int question. Why float64 and not float32?

File: src/generator/protocol/operations.ts (hunk 1 above)
Comment (jhendrixMSFT) [reply src/generator/protocol/operations.ts]: `strconv.FormatInt()` takes an `int64` as its first parameter so we have to up-cast in this case (same is true for `FormatFloat` below).

Comment (catalinaperalta) [reply src/generator/protocol/operations.ts]: Ah I see, thanks for the explanation!

File: test/autorest/urlgroup/queries_test.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+func TestArrayStringCsvValid(t *testing.T) {
+	client := getQueriesClient(t)
+	result, err := client.ArrayStringCsvValid(context.Background(), []string{
+		"ArrayQuery1", "begin!*'();:@ &=+$,/?#[]end", "", "",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if result.StatusCode != http.StatusOK {
```
Comment (catalinaperalta) [inline test/autorest/urlgroup/queries_test.go]: This is just a question, why did you switch from using the deep equals func to just checking the status code? It's the same result for the successful status codes, so it doesnt really matter, but just wondering why you preferred this.

Comment (jhendrixMSFT) [reply test/autorest/urlgroup/queries_test.go]: It was shorter to write since there is no response body to compare, just the status code.

**Pull request merged**